tests/database/test_database.py

```diff
@@ -0,0 +1,150 @@
+import pytest
+from unittest.mock import patch
+from python.exceptions import (
+    DuplicateRecordError,
+    GetUserFromIdError,
+    GetIdFromEmailError,
+)
+from datetime import datetime
+
+
+class TestGetIdFromEmail:
+    """
+    Test the `get_id_from_email` function for both success and failure cases.
+    """
+
+    def test_get_id_from_email_success(self, access_db, create_user):
+        """Test that the correct user ID is retrieved when the email exists."""
+
+        # Try to get the user ID based on the email
+        user_id = access_db.get_id_from_email(email="testuser@example.com")
+
+        # Assert that the correct user ID is returned
+        assert user_id == 1
+
+    def test_get_id_from_email_not_found(self, access_db):
+        """Test that a GetIdFromEmailError is raised when the email does not exist."""
+
+        # Test that the exception is raised
+        with pytest.raises(GetIdFromEmailError) as exc_info:
+            access_db.get_id_from_email(email="testuser@example.com")
+
+        # Check the exception message
+        assert str(exc_info.value) == "User with email testuser@example.com not found"
```

Comment:
```suggestion
    def test_get_id_from_email_not_found(self, access_db, create_user):
        """Test that a GetIdFromEmailError is raised when the email does not exist."""

        # Test that the exception is raised
        with pytest.raises(GetIdFromEmailError) as exc_info:
            access_db.get_id_from_email(email="fakeuser@example.com")

        # Check the exception message
        assert str(exc_info.value) == "User with email fakeuser@example.com not found"
```
I think its clearer if we create the test user, then search for a fake user, instead of not creating a test user and then searching for it.

Reply:
Resolved in [d91ce60](https://github.com/rowan04/L5-SE-CW2-QualityPlan/pull/14/commits/d91ce60cb4fba350a53d8e52a5b499b8a095c5d1)